simpleviewinc/keg-herkin
# Zen 548 load file

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-548)

## Context
* PR Pt 2
* we need to be able to load the files selected in the sidebar to the code editor
* ensure that if it's a feature file it will load in the feature editor alongside the definitions editor
* if it's not a feature file, load it with the default editor
* close sidebar on file select

## Goal

* close sidebar on file select
* load mounted files into the code editor with the proper editor type

## Updates

* `repos/tap/actions/files/setActiveFile.js`
   * set selected file as active file
* `repos/tap/components/codeEditor/codeEditor.js`
  * update to make the editors general
  * migrated Feature / Definitions specific logic onto the `FeatureEditor` / `DefinitionsEditor` respectively 
* `repos/tap/components/feature/featureEditor.js`
  * moved over some of the feature logic from `CodeEditor`
* `repos/tap/components/definition/definitionsEditor.js`
  * moved over some of the definitions logic from `CodeEditor`  
* `repos/tap/containers/rootContainer.js`
  * added prop drilling to be able to toggle the sidebar on/off based on state
* `repos/tap/hooks/useFeature.js`
  * updated this hook to take in either a name or path
  
## Testing

### Setup
* Pull this branch
* Run `keg herkin && keg herkin start --config ./repos/example/herkin.config.js --from keg-herkin:zen-548-load-file`
* navigate to http://herkin-zen-548-load-file.local.kegdev.xyz/

### Test 1
* Dismiss the modal by Pressing Do It on the modal button (doesn't matter what options you pick
* Toggle open the side menu
* Verify you see bdd and waypoint dropdown
* Drill into `/bdd/features` and select `examples.feature`
* Verify that the file gets loaded onto the editor along with gherkin styling and definitions editor
  * [something like this](https://user-images.githubusercontent.com/3317835/108408508-5787d780-71e2-11eb-8e5e-4a077f14ae4e.png)
* Open the side menu again
* Drill into `/waypoint` and select `foo.test.js`
* Verify it updates the default editor with the javascript content of `foo.test.js` and the definitions editor is hidden
 * verify that file updates are reflected in both the `Code Editor` and `Test Runner` tab
